packages/amplify-category-storage/provider-utils/awscloudformation/service-walkthroughs/s3-walkthrough.js

```diff
@@ -885,13 +886,13 @@ async function addTrigger(context, resourceName, triggerFunction, adminTriggerFu
       };
 
       // Update DependsOn
-      context.amplify.updateamplifyMetaAfterResourceUpdate(category, resourceName, 'dependsOn', [
-        {
-          category: 'function',
-          resourceName: functionName,
-          attributes: ['Name', 'Arn', 'LambdaExecutionRole'],
-        },
-      ]);
+      const dependsOnResources = amplifyMetaFile.storage[resourceName].dependsOn;
```

Comment:
Would depends on always be a array? Is it possible that this could be `undefined` or `null`